packages/components/pie-modal/test/component/pie-modal.spec.ts

```diff
@@ -83,10 +93,144 @@ test.describe('`Pie Modal is closed`', () => {
                 },
             );
 
+            // Act
             await page.locator('#dialog').click();
 
+            // Assert
             expect(messages).toHaveLength(1);
         });
     });
 });
 
+test.describe('Pie Modal `isDismissible`', () => {
+    test.describe('when `isDismissible` is truthy', () => {
+        test('should contain a close button so the modal can be dismissed', async ({ mount }) => {
+            // Arrange
+            const component = await mount(
+                PieModal,
+                {
+                    props: {
+                        isOpen: true,
+                        isDismissible: true,
+                    },
+                },
+            );
+
+            // Act & Assert
+            await expect(component.locator('.c-modal-closeBtn')).toBeVisible();
+        });
+
+        test('should close the modal when the close button is clicked', async ({ mount }) => {
+            // Arrange
+            const component = await mount(
+                PieModal,
+                {
+                    props: {
+                        isOpen: true,
+                        isDismissible: true,
+                    },
+                },
+            );
+
+            // Act
+            await component.locator('.c-modal-closeBtn').click();
+
+            // Assert
+            await expect(component).not.toBeVisible();
+        });
+
+        test('should close the modal when the backdrop is clicked', async ({ mount }) => {
+            // Arrange
+            const component = await mount(
+                PieModal,
+                {
+                    props: {
+                        isOpen: true,
+                        isDismissible: true,
+                    },
+                },
+            );
+
+            // Act
+            await component.locator('dialog').click();
```

Comment:
I'm assuming that this clicks on the backdrop and not the modal then? Do we know why this works?

Reply:
AFAIK you can't select a pseudo element. I saw there is a way to evaluate the page and pass in the pseudo element but im not sure that helps with the actual clicking part. Saying that though i think there is something wrong with this test as it seems to pass in both cases. I will double check it.

Reply:
I've updated them now. There isn't really a good way to selector the backdrop (in playwright). I tried targetting the `open` attribute which toggles but playwright doesn't have an API to check the attribute exists only the value of the attribute so felt hacky. I think the changes i added now works best imo now.

Reply:
What would the value of the attribute be if it doesn't exists @kevinrodrigues? Could that potentially allow us to infer?